### PR TITLE
Init without working range size

### PR DESCRIPTION
**Pull request by BasThomas**
## Changes in this pull request

Issue fixed: #414 

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [ ] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

What should we do with testing? I changed all the occurrences of the old initializer with the new one, so I think this is good to go?

### CODE REVIEW

**Comment by BasThomas** [comment]
@ocrickard, @rnystrom thought you might be interested in looking at this as well.

File: Source/IGListAdapter.h [hunk 1]
```diff
@@ -103,6 +103,17 @@ IGLK_SUBCLASSING_RESTRICTED
                workingRangeSize:(NSInteger)workingRangeSize NS_DESIGNATED_INITIALIZER;
 
 /**
+ Initializes a new `IGListAdapter` object.
```
Comment (rnystrom) [inline Source/IGListAdapter.h]: Let's add that this uses workingRangeSize 0

Comment (BasThomas) [inline Source/IGListAdapter.h]: Also changed the opt-out to prefer the new init over providing `0` as a value.

File: Source/IGListAdapter.h [hunk 2]
```diff
@@ -96,13 +96,26 @@ IGLK_SUBCLASSING_RESTRICTED
  the previous and succeeding 2 objects will be notified that they are within the working range. As you scroll the list
  the range is updated as objects enter and exit the working range.
 
- To opt out of using the working range, you can provide a value of `0`.
+ To opt out of using the working range, use `initWithUpdater:viewController` instead.
```
Comment (rnystrom) [inline Source/IGListAdapter.h]: Suuuuuper nit: can this be

> To opt out of using the working range, use `initWithUpdater:viewController:` or provide a working range of `0`.

Comment (BasThomas) [inline Source/IGListAdapter.h]: Of course!

**Pull request closed without merging**